alpa/util.py

```diff
@@ -1243,3 +1255,247 @@ def try_import_ray_worker(error: bool = False):
+
+    ray_state = try_import_ray_state()
+    resources_list = ray_state.state._available_resources_per_node(  # pylint: disable=protected-access
+    ).values()
+
+    for resource in resources_list:
+        resource_name_list = resource.keys()
+
+        node_ip = None
+        bundle_index_list = []
+        for resource_name in resource_name_list:
+            # when bundles are created, pg resources are
+            # specified as [resource]_[bundle_index]_[pg_id]
+            if pg:
+                try_bundle_index = re.findall(rf"bundle_group_(\d+)_{pg_id}",
+                                              resource_name)
+            else:
+                try_bundle_index = re.findall(r"bundle_group_(\d+)_.*",
+                                              resource_name)
+
+            try_node_ip = re.findall(
+                r"^node:(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$)", resource_name)
+
+            if try_node_ip:
+                node_ip = try_node_ip[0]
+
+            if try_bundle_index:
+                bundle_index_list.append(try_bundle_index[0])
+
+        dict_bg2ip.update(
+            **dict(zip(bundle_index_list, [node_ip] * len(bundle_index_list))))
+
+    ip_list = []
+    for i in range(len(dict_bg2ip)):
+        ip_list.append(dict_bg2ip[str(i)])
+
+    return ip_list
+
+
+def env_integer(key, default):
+    if key in os.environ:
+        value = os.environ[key]
+        if value.isdigit():
+            return int(os.environ[key])
+
+        logger.debug(f"Found {key} in environment, but value must "
+                     f"be an integer. Got: {value}. Returning "
+                     f"provided default {default}.")
+        return default
+    return default
+
+
+def create_placement_group(num_hosts,
+                           num_devices_per_host,
+                           additional_resources_per_host=None):
+    """Creates a placement group if it does not exist.
+
+    If a placement group is already detected (Tune) this will be a no-op.
+
+    By default the placement group will be created with `SPREAD` strategy.
+    This is optimized for colocating GPUs on different nodes.
+
+    If a placement group is created it will be return the current
+    placement group
+
+    Args:
+        num_hosts: the number of hosts to create the placement group for
+        num_devices_per_host: the number of devices per host
+        additional_resources_per_host: additional resources per host
+
+    Returns:
+        The placement group
+    """
+    current_placement_group = get_current_placement_group()
+    ray_worker = try_import_ray_worker()
+    worker = ray_worker.global_worker  # pylint: disable=protected-access
+    should_capture_child_tasks_in_placement_group = (
+        worker.should_capture_child_tasks_in_placement_group)
+    should_create_placement_group = (
+        current_placement_group is None or
+        not should_capture_child_tasks_in_placement_group)
+
+    if should_create_placement_group:
```

Comment:
`should_create_placement_group` is always True when using alpa along. `should_create_placement_group` can be false when using Tune integration. Is this correct?